functions.sh

```diff
@@ -336,6 +336,19 @@ nameservers() {
   dig +short ns "${1}" | sort | tr "[:upper:]" "[:lower:]"
 }
 
+realpath() {
+  if [[ -z "${1}" ]]; then
+    return 1
+  fi
+
+  if ! type -P realpath; then
+    python -c "import os; print(os.path.realpath('${1}'))"
```

Comment:
I don't want to carry a Python requirement. Just do this:

```
realestpath() {
    echo "$(cd $(dirname $1); pwd)/$(basename ${1})"
}
```

Reply:
Did you **_actually_** test this? That's not going to work right in many cases, but Python (which is bundled onto every system we operate) will:

```bash
dilly@lappy refactored-octo-doodle $ ( set -x; command realpath /usr/local/bin/jq )
+ command realpath /usr/local/bin/jq
+ realpath /usr/local/bin/jq
/usr/local/Cellar/jq/1.6/bin/jq    # Correct


dilly@lappy refactored-octo-doodle $ ( set -x; realestpath /usr/local/bin/jq )
+ realestpath /usr/local/bin/jq
+++ dirname /usr/local/bin/jq
++ cd /usr/local/bin
++ pwd
++ basename /usr/local/bin/jq
+ echo /usr/local/bin/jq
/usr/local/bin/jq    # Incorrect


dilly@lappy refactored-octo-doodle $ python -c "import os; print(os.path.realpath('/usr/local/bin/jq'))"
/usr/local/Cellar/jq/1.6/bin/jq    # Correct
```

Here's a [good thread](https://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac) about why "simple" problems often have complicated solutions. You should review it for context on this problem in particular.